src/store/actions.js

```diff
@@ -1,70 +1,123 @@
+      dispatch('_fetchListById', currentListId);
+    }
   },
-  _addList({ commit, dispatch }, list) {
-    commit('addList', list);
-
-    dispatch('_updateLocal');
+  _updateListIdInLocalStorage({ getters }) {
+    localStorage.setItem('currentListId', JSON.stringify(getters.currentListId));
   },
-  _saveList({ commit, dispatch }, list) {
-    commit('saveList', list);
 
-    dispatch('_updateLocal');
+  // lists
+
+  _fetchListsForUser({ commit, dispatch }) {
+    return this._vm.$axios.get(`${this._vm.$apiBasePath}lists`)
+      .then(response => {
+        commit('setLists', response.data);
+        dispatch('_setListIdFromLocalStorage');
+      })
+      .catch(error => {
+        throw error;
+      });
   },
-  _removeList({ commit, dispatch, getters }, listId) {
-    if (getters.currentListId === listId) {
-      const switchListId = getters.lists.length > 1
-        ? getters.lists.find(list => list.id !== listId).id
-        : null;
+  _fetchListById({ commit, dispatch, getters }, id) {
+    dispatch('_setCurrentListId', id);
 
-      commit('switchList', switchListId);
+    if (getters.currentListObj.items.length) {
+      if (getters.currentListObj.items[0] instanceof Object) {
+        dispatch('_setCurrentItems', getters.currentListObj.items);
+      }
     }
 
-    commit('removeList', listId);
-
-    dispatch('_updateLocal');
+    return this._vm.$axios.get(`${this._vm.$apiBasePath}list/${id}`)
+      .then(response => {
+        commit('updateList', response.data);
+        dispatch('_setCurrentItems', response.data.items);
+      })
+      .catch(error => {
+        throw error;
+      });
   },
-  _setListForEditting({ commit }, id) {
-    commit('setListForEditting', id);
+  _addList({ commit, dispatch }, list) {
+    return this._vm.$axios.post(`${this._vm.$apiBasePath}list/add`, list)
+      .then(response => {
+        commit('addList', response.data);
+        dispatch('_setCurrentListId', response.data.id);
+      })
+      .catch(error => {
+        throw error;
+      });
   },
-  _filterList({ commit, dispatch, getters }, filters) {
-    commit('filterList', {
-      currentListIndex: getters.currentListIndex,
-      filters,
-    });
+  async _addTestList({ commit, dispatch }, listToSend) {
+    const listDataToSend = {
+      name: listToSend.name,
+      isPrivate: listToSend.isPrivate,
+      tags: listToSend.tags,
+      categories: listToSend.categories,
+    };
+    const listResponse = await this._vm.$axios.post(`${this._vm.$apiBasePath}list/add`, listDataToSend);
+    const list = listResponse.data;
 
-    dispatch('_updateLocal');
+    commit('addList', list);
+    dispatch('_setCurrentListId', list.id);
+
+    const itemsResponse = await this._vm.$axios
+      .post(`${this._vm.$apiBasePath}items/add-many/${list.id}`, { items: listToSend.items });
+    const items = itemsResponse.data;
+
+    commit('addItems', items);
+    dispatch('_setCurrentItems', items);
+  },
```

Comment:
but about async/await - great!
I think we can change it after everywhere where we can do it